### PR TITLE
Implement read() and write() function

### PR DESCRIPTION
**Pull request by DK-DARKmatter**
Implement read() and write() function, tls part needs to be
implemented in the future.

### CODE REVIEW

**Comment by DK-DARKmatter** [comment]
Why the `cargo test --verbose` failed?

File: src/lib.rs [hunk 1]
```diff
@@ -194,13 +194,21 @@ pub extern "C" fn socket(domain: c_int, socktype: c_int, protocol: c_int) -> c_i
...
-// fd: c_int,
-// buf: *const c_void,
-// count: usize,
-// ) -> isize {
-//}
+#[no_mangle]
+pub extern "C" fn write(fd: c_int, buf: *const c_void, count: usize) -> isize {
+    match INDEX.get(fd) {
+        None => return next!(write(fd, buf, count)),
+        Some(s) => 0, //TODO: tls_write() implement
```
Comment (npmccallum) [inline src/lib.rs:201]: Never return success in a stubbed in method. Always return an error.

File: src/lib.rs [hunk 2]
```diff
@@ -194,13 +194,21 @@ pub extern "C" fn socket(domain: c_int, socktype: c_int, protocol: c_int) -> c_i
...
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn read(fd: c_int, buf: *mut c_void, count: usize) -> isize {
+    match INDEX.get(fd) {
+        None => return next!(read(fd, buf, count)),
+        Some(s) => 0, //TODO: tls_read() implement
```
Comment (npmccallum) [inline src/lib.rs:209]: Same as above.

**Review by npmccallum — CHANGES_REQUESTED** [review]

**Comment by npmccallum** [comment]
> Why the `cargo test --verbose` failed?

Probably because the test process calls `read()` and/or `write()` and you changed the behavior subtly. Alternatively, the `next!()` macro is broken.

**Comment by DK-DARKmatter** [comment]
Change the name of the function indeed pass the test. But that's not the thing we expect. Is there any other alternative way we could solve this?

**Comment by npmccallum** [comment]
You need to debug the problem.

**Comment by npmccallum** [comment]
Obsoleted by #21.

**Pull request closed without merging**